### ISSUE TITLE
False positive `UnknownArgumentType` in loop

### ISSUE THREAD

**Issue by Molkree** [microsoft/pyright#3676]
**Describe the bug**
False positive `UnknownArgumentType` in loop.

**To Reproduce**
Don't mind the meaning of the code, it's just a minimal example I could figure out.
```py
def foo(nums: list[int], target: int) -> None:
    var = nums[0]
    while True:
        mid = var  # no error if I explicitly type `mid` here as `mid: int = var`
        if nums[mid] == target:  # Argument type is unknown/Argument corresponds to parameter "__i" in function "__getitem__"
            return
        if var:
            var = mid
        else:
            var = mid
```

**Expected behavior**
No errors.

**Screenshots or Code**
Curiously, `mid` has correct type in the tooltip.

![image](https://user-images.githubusercontent.com/12860284/177862997-545488e2-7701-450f-aebb-adefc701f732.png)

**VS Code extension or command-line**
I'm coming here as a Pylance user and it's present in the latest stable (2022.7.20) and preview (2022.7.21) release of Pylance. But I have checked using just the Pyright extension and it seems like it's a regression from [1.1.234](https://github.com/microsoft/pyright/releases/tag/1.1.234) release of Pyright.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.258, which I just published. It will also be included in the next release of pylance.
